controllers/cnbi/controller.go

```diff
@@ -55,47 +52,37 @@ type CustomNBImageReconciler struct {
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
 func (r *CustomNBImageReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+
 	logger := log.FromContext(ctx)
-	r.CNBi = &meteorv1alpha1.CustomNBImage{}
+	CNBi := meteorv1alpha1.CustomNBImage{}
 
-	if err := r.Get(ctx, req.NamespacedName, r.CNBi); err != nil {
+	if err := r.Get(ctx, req.NamespacedName, &CNBi); err != nil {
 		if errors.IsNotFound(err) {
-			logger.Info("Resource deleted")
-			return ctrl.Result{}, nil
+			logger.Info("Resource was deleted.")
+			err = nil
 		}
-		logger.Error(err, "Unable to fetch reconciled resource")
-		return ctrl.Result{Requeue: true}, err
+		return ctrl.Result{}, err
```

Comment:
I think you are merging two different errors here: 1. resource is not found, and 2. error getting the resource. 2. should yield a requeing of the reconcile.

Reply:
We don't need to explicitly Requeue, if `err` is not `nil` a Requeue is triggered. Using err = nil instead of explicitly returning with nil is a matter of taste though.